### PR TITLE
fix(j-s): Registrar Indictment Completion

### PR DESCRIPTION
**Pull request by gudjong**
# Registrar Indictment Completion

[Dómritarar geta ekki lokið ákærum](https://app.asana.com/0/1199153462262248/1203705883653532/f)

## What

- Adds an optional callback to roles rules given the rules access to the request.
- Modifies registrar transition rule to allow registrars to complete indictment cases.
- Creates a new defender file rule to limit defender case file access to indictment ruling only.

## Why

- Bug fixes and improved security.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/backend/src/app/modules/file/test/fileController/updateFilesGuards.spec.ts [hunk 1]
```diff
@@ -8,6 +9,7 @@ import {
   CaseWriteGuard,
 } from '../../../case'
 import { CaseTypeGuard } from '../../../case/guards/caseType.guard'
+import { CaseFileExistsGuard } from '../../guards/caseFileExists.guard'
```
Comment (oddsson) [inline apps/judicial-system/backend/src/app/modules/file/test/fileController/updateFilesGuards.spec.ts:12]: This is never used, please remove it.

File: libs/judicial-system/auth/src/lib/auth.types.ts [hunk 2]
```diff
@@ -6,21 +6,37 @@ export type Credentials = {
 }
 
 export enum RulesType {
+  BASIC,
   FIELD,
   FIELD_VALUES,
 }
 
-export type RolesFieldRule = {
+type RolessBasicRule = {
```
Comment (oddsson) [inline libs/judicial-system/auth/src/lib/auth.types.ts:14]:
```suggestion
type RolesBasicRule = {
```

File: libs/judicial-system/auth/src/lib/auth.types.ts [hunk 3]
```diff
@@ -6,21 +6,37 @@ export type Credentials = {
...
   dtoField: string
   dtoFieldValues: string[]
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  canActivate?: (request: any) => boolean
 }
 
-export type RolesRule = UserRole | RolesFieldRule | RolesFieldValuesRule
+export type RolesRule =
+  | UserRole
+  | RolessBasicRule
```
Comment (oddsson) [inline libs/judicial-system/auth/src/lib/auth.types.ts:40]:
```suggestion
  | RolesBasicRule
```

**Review by oddsson — CHANGES_REQUESTED** [review]
Just a couple of minor things

File: libs/judicial-system/auth/src/lib/auth.types.ts [hunk 4]
```diff
@@ -6,21 +6,37 @@ export type Credentials = {
...
   FIELD,
   FIELD_VALUES,
 }
 
-export type RolesFieldRule = {
+type RolessBasicRule = {
+  role: UserRole
+  type: RulesType.BASIC
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  canActivate?: (request: any) => boolean
```
[the remark this answers is not in the document]
Comment (gudjong) [reply libs/judicial-system/auth/src/lib/auth.types.ts:18]: Not really as then I need to use the dreaded `as` in each `canActivate` method defined in the roles rules. I struggled with this a bit and then gave up.

**Review by oddsson — APPROVED** [review]

**Pull request merged**